### PR TITLE
Added fill_diagonal to numpy frontend

### LINKED ISSUE #13386: fill_diagonal

### PR DESCRIPTION
**Pull request by Aneesh02**
Closes #13386

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py [hunk 1]
```diff
@@ -190,3 +190,43 @@ def test_indices(
...
+        dimensions=helpers.get_shape(),
+        dtype = helpers.get_dtypes("valid", full=False),
+        sparse=helpers.get_dtypes(kind="bool"),
+        val=helpers.ints(min_value=1, max_value=10),
+        ndim=helpers.ints(min_value=2, max_value=10),
+        test_with_out=st.just(False),
+    )
+    def test_fill_diagonal(
+            *,
+            dimensions,
```
Comment (sherry30) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py:215]: I think you forgot to add `dtype_x_indices_axis` which you are generating in the decorator above

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py [hunk 2]
```diff
@@ -190,3 +190,43 @@ def test_indices(
...
+    ):
+        helpers.test_frontend_function(
+            input_dtypes=dtype,
+            test_flags=test_flags,
+            frontend=frontend,
+            fn_tree=fn_tree,
+            on_device=on_device,
+            dimensions=dimensions,
+            dtype=dtype[0],
+            sparse=sparse,
```
Comment (sherry30) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py:231]: Not sure why you're passing these arguments 🤔 
The arguments accepted by your frontend functions are `a`, `val`, and `wrap`. These are the ones that should be passed here

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

There are a few lint errors in the files you changed, can you fix those.


Thanks

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py (hunk 1 above)
Comment (Aneesh02) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py]: Yes I wanted to use dtype_and_x

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py (hunk 2 above)
Comment (Aneesh02) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py]: Yes I was figuring out how to write the correct unit tests. Thanks for pointing it out.

**Comment by Aneesh02** [comment]
Can you please help me resolve the new issue that is coming up? How can I do assignment in array values? The following is not supported `a.flatten[:end:step] = val` . Thanks

**Comment by Aneesh02** [comment]
I have been trying to resolve the issues. Looks like I need a little help on how to proceed further. @sherry30 request you help me resolve the errors.

**Comment by Aneesh02** [comment]
Okay, sure I will try to debug this thanks for the review.

**Comment by Aneesh02** [comment]
Working on it!

**Comment by Aneesh02** [comment]
Still working on it!

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**